src/hooks/useTheme.js

```diff
@@ -12,24 +12,32 @@ const createAntStylesheet = () => {
   document.head.appendChild(antStylesheet);
 };
 
-export default function useTheme() {
-  const darkTheme = window.matchMedia('(prefers-color-scheme: dark)').matches;
-  const [theme, setTheme] = useLocalStorage(
-    'theme',
-    darkTheme ? 'dark' : 'light'
-  );
-  const toggleTheme = () => setTheme(theme === 'dark' ? 'light' : 'dark');
+const checkMode = () => {
+  const theme = localStorage.getItem('theme');
+  if (theme === null) {
+    if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
```

Comment:
Great job on adding this theme persistence feature! I love your implementation checks for user theme preference as well.